Source/WebCore/rendering/svg/RenderSVGInlineText.cpp

```diff
@@ -221,16 +221,19 @@ void RenderSVGInlineText::computeNewScaledFontForStyle(const RenderObject& rende
 
     // Alter font-size to the right on-screen value to avoid scaling the glyphs themselves, except when GeometricPrecision is specified
     scalingFactor = computeScalingFactor();
-    if (!scalingFactor || style.fontDescription().textRenderingMode() == TextRenderingMode::GeometricPrecision) {
+    if (scalingFactor == 1 || !scalingFactor) {
```

Comment:
The comment is out of sync now.

Out of curiosity: I wonder if we have coverage for the scalingFactor=0 condition -- just thinking loudly if we can arrive at that condition (implying that 'Screen CTM' from <text> children up to <svg> root, has zero-scale components... can we arrive there without explicitely constructing a matrix that has e.g. sx/sy = 0....)

Reply:
Out of sync in terms of placement not content, the GeometricPrecision condition described in the comment is now checked elsewhere...

The bottom line of the patch is: we previously did not apply any scaling when GeometricPrecision was specified. That means we correctly ignored the scaling due to the "screen ctm" (as one is supposed to stay at the same font-size in geo-precision mode all the time, even when scaling), however we failed to go through Style::computedFontSize... to recompute the font-size, ignoring any 'zoom' induced scaling (and minimum font size rules, etc.)

Correct?

Reply:
It was raised by @shallawa as well but I checked on Chrome Source and they still have this code ['scalingFactor = 1'](https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/layout/svg/layout_svg_inline_text.cc;l=496). So should I remove the second portion splitting (style.fontDescription().textRenderingMode() == TextRenderingMode::GeometricPrecision)?

Reply:
You mean scalingFactor is reset to 1 if it was zero? Sure, you can always clamp, I'd just like to understand if we're expecting valid situations where we enter the method with scalingFactor=0 - that's why I asked for test coverage.

I never looked at Blink source code to guide a decision, but rather try to come up with a solution from first principles, best baked by spec references etc. But that might be caused be me being a physicist :-) If you compare with other source code, keep in mind that their underlying platform constraints might be totally different, and they might need protections for things we don't or vice-versa.

Reply:
True - sometime we need to think from first principle. Although, honestly I might not be right person to answer your queries since I am novice to C++ and Webkit source code. I just came across this easy to merge fix but I am not able to tackle cases beyond this simple merge. Although, Blink also did this change - https://bugs.webkit.org/show_bug.cgi?id=245356, to tackle out of range scenarios especially for test cases about SVG zoom-in. I am happy to be guided.

Reply:
> True - sometime we need to think from first principle. Although, honestly I might not be right person to answer your queries since I am novice to C++ and Webkit source code. I just came across this easy to merge fix but I am not able to tackle cases beyond this simple merge. Although, Blink also did this change - https://bugs.webkit.org/show_bug.cgi?id=245356, to tackle out of range scenarios especially for test cases about SVG zoom-in. I am happy to be guided.

It does look like you're on a fast track you become knowledgeable in these areas, keep on :-)
The other bug report is also interesting, I commented there as well.